### PR TITLE
Feature/ont 36

### PR DESCRIPTION
**Pull request by dmeidlin**
User Story
As a user
I want to be able to see the list of traits that exist in the breeding program

Design Reference
Trait Basic Actions

Acceptance Criteria
Given: The program has one or more traits defined
When: the user has navigated to the "Trait Management" page
Then: ensure that the list of traits that exist within the program are displayed in a table

Design reference: page 2

Given: The program does not have any traits currently defined
When: the user has navigated to the "Trait Management" page
Then: ensure that an empty table DOES NOT display
And: ensure that a message displays indicating to the user that there are no traits for the program, and new traits can be created by clicking "New Trait" or "Import Traits"

Design reference: similar in design to empty table for program locations

Technical Notes
If ONT-47 - API - Create endpoints to fetch traits isn't done when this card is being implemented, services/daos in the UI that are created can return static (dummy) data. Once ONT-47 is complete, the services/daos can be updated to call to the API.

NOTE: The "Show Details" link was left off since this part of the component will likely change as part of ONT-37.

### CODE REVIEW

**Comment by dmeidlin** [comment]
[ONT-36 - View list of traits](https://trello.com/c/3tdjzSRn/221-ont-36-view-list-of-traits)

File: src/views/TraitsList.vue [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  - Unless required by applicable law or agreed to in writing, software
+  - distributed under the License is distributed on an "AS IS" BASIS,
+  - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  - See the License for the specific language governing permissions and
+  - limitations under the License.
+  -->
+
+<template>
+  <div class="traits-list">
+    <TraitListsTable
```
Comment (nickpalladino) [inline src/views/TraitsList.vue:20]: Missing the "All Traits" title text above the table as shown in the design reference.

File: src/components/trait/TraitListsTable.vue [hunk 2]
```diff
@@ -0,0 +1,279 @@
...
+      </section>
+      <div class="columns">
+        <div class="column is-whole has-text-centered buttons">
+          <button v-on:click="modalDeleteHandler()" class="button is-danger"><strong>Yes, remove</strong></button>
+          <button v-on:click="deactivateActive = false" class="button">Cancel</button>
+        </div>
+      </div>              
+    </WarningModal>
+
+    <button
```
[the remark this answers is not in the document]
Comment (eawoods) [reply src/components/trait/TraitListsTable.vue:38]: It's good to have the new-trait button and form are here as placeholders to review placement, I think.  
Nick is right that both of these elements will need to be hidden or removed to ship a 0.1 release that doesn't include the ability to add traits from the UI as currently planned.

File: src/views/TraitsList.vue (hunk 1 above)
Comment (dmeidlin) [reply src/views/TraitsList.vue:21]: Thanks, the All Traits h1 title is in place, now.

File: src/components/trait/TraitListsTable.vue (hunk 2 above)
[the remark this answers is not in the document]
Comment (dmeidlin) [reply src/components/trait/TraitListsTable.vue]: Makes sense. I removed the New Trait button.

[the remark this answers is not in the document]
Comment (dmeidlin) [reply src/components/trait/TraitListsTable.vue]: ...and I updated the trello card.

**Comment by nickpalladino** [comment]
Thanks, changes look good. You want to take care of the merge conflicts so we can make sure it's all good?

**Comment by dmeidlin** [comment]
> Thanks, changes look good. You want to take care of the merge conflicts so we can make sure it's all good?

Yep, I'll take care of that. Thanks!

**Pull request merged**